### PR TITLE
feat: Notification menu

### LINKED ISSUE #400: Add a notifications button
When installing mod PRs, etc i quite often alt+tab to read the PR descriptions and miss the notification that the PR has been successfully installed. While im sure your code is robust and wouldn't fail, it would be a nice QOL feature to be able to check previous notifications.

Something, something like this:
image

### PR DESCRIPTION
**Pull request by Alystrasz**
This PR introduces a notification menu, which hosts notifications fired while the app was not focused.
Notifications can be closed by the user.

Closes #400.

##### Screenshots

image

image

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Ooooh, me likey 👀

File: src-vue/src/App.vue [hunk 1]
```diff
@@ -34,9 +36,6 @@ export default {
     this.$root!.$i18n.locale = lang;
   },
   methods: {
-    async toggleMaximize() {
-      await appWindow.toggleMaximize();
-    },
```
Comment (GeckoEidechse) [inline src-vue/src/App.vue:39]: Is the reason for removing maximise to make space for the notification icon? ^^

Comment (Alystrasz) [reply src-vue/src/App.vue:39]: Yep, and also because I don't like the icon + window can still be maximized by double-clicking the menu bar ;)

**Review by GeckoEidechse — COMMENTED** [review]
So far worked fine in initial testing :D

**Comment by GeckoEidechse** [comment]
Tagging @F1F7Y for review as well as they requested the feature

File: src-vue/src/App.vue (hunk 1 above)
Comment (GeckoEidechse) [reply src-vue/src/App.vue:39]: Aight, in that case I'll just make a separate PR with that change rq to get the diff down a bit more ^^

Comment (GeckoEidechse) [reply src-vue/src/App.vue:39]: #617

**Review by GeckoEidechse — APPROVED** [review]
Worked fine in testing for me. Pressed a bunch of buttons that trigger notifications and alt-tabbed before they displayed to get said notifications sent to notification icon.

File: src-vue/src/utils/ui.ts [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+ * If the app is not focused when this is invoked, a notification is added to the notifications menu.
  **/
 function showNotification(
     title: string,
     message: string = '',
     type: 'success' | 'warning' | 'error' | 'info' = 'success',
     duration: number = 4500
 ): NotificationHandle {
+    if (!document.hasFocus()) {
+        store.commit('addNotification', {title, text: message, type});
```
Comment (GeckoEidechse) [inline src-vue/src/utils/ui.ts:16]: I think there's probably also something in the Tauri API to make the taskbar icon flash, although I'm fine with moving that feature to a separate PR \^\^

**Review by GeckoEidechse — APPROVED** [review]
Code looks fine as far as I can tell. Didn't see any obvious issues or the like \^\^

File: src-vue/src/utils/ui.ts (hunk 2 above)
Comment (GeckoEidechse) [reply src-vue/src/utils/ui.ts:16]: Moved to #634
